### PR TITLE
[processor/transform] Add IsMatch function

### LINKED ISSUE #11077: [processor/transform] Add 'like' capability to WHERE condition
**Is your feature request related to a problem? Please describe.**
The transform processor's where condition cannot handle checking if 2 string match only in specific positions.  It must check the whole string. 

**Describe the solution you'd like**
For strings, a condition allows you to use the keyword `like` which will use glob matching to compare the strings.  There should also be a way to negate the match, such as `not like`.

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
ThisPR adds a new factory function called `IsMatch`.  This function performs regular expression matching against a string and returns a bool.  The intended purpose is to be able to use regular expression matching for conditions, but it could be used other places as well.

**Link to tracking Issue:** 
Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11077

**Testing:** 
Added new tests

**Documentation:** 
Updated readme

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

File: processor/transformprocessor/README.md
Comment (TylerHelmuth) [reply processor/transformprocessor/README.md:29]: I like sticking with our function syntax standardss, which says Factories don't need to be a verb but regular functions (maybe we should formalize a name for these functions, maybe transformers?) must be a verb.  Since IsMatch is a factory I like it not starting with a verb to avoid confusion with transformers.

**Comment by TylerHelmuth** [comment]
@open-telemetry/collector-contrib-approvers please review.

**Review by djaglowski — APPROVED** [review]
LGTM, provided conflicts are resovled

**Pull request merged**
